src/ext/oblivc/obliv_bits.c

```diff
@@ -1842,6 +681,7 @@ void feedOblivInputs(OblivInputs* spec, size_t count, int party)
     { __obliv_c__##ot rv; \
       OblivInputs spec; \
       setupObliv##tname(&spec,&rv,v); \
+      spec.src_f = v; \
```

Comment:
In memory, floats and ints are stored in entirely different formats (as I'm sure you know, since you did just write an entire codebase dealing with float-formatted data). Since the type of `spec.src_f` is float, I believe it is the case that `v` will be implicitly converted to a float here (remember, `v` is integral in all but one instantiation of this function), and then the memory (or at least the first 32 bits of it) underlying the union of `spec.src_f` and `spec.src` will be overwritten with float-formatted data, regardless of the input type. There is definitely an overwrite happening, because you're performing an assignment. The _value_ stored in `spec.src` may be the same, but the _representation_ will be different, and this is important.  Your float debugging protocol always reads `spec.src_f`, and therefore will print some value that looks correct (plus/minus representational errors that happen when converting to float) regardless of whether the input was an int or a float. The debug protocol, on the other hand, reads `spec.src` (see `dbgProtoFeedOblivInputs`). Because `spec.src` is unioned with `spec.src_f`, no type conversion will be done during this read - rather, the raw float-formatted bits will be interpreted as an integer (which is wrong). Have you tried the debug protocol? Regardless of whether there is some confounding factor here that makes it somehow work out correctly, my interpretation is that `spec.src` is meant to hold the incoming data in its original form, which means that it should _not_ be converted to or stored as a float unless it was one to begin with.

Maybe I have missed something here - if so, please let me know. For the record, I haven't actually run the debug protocol with your changes yet, but I can, if it helps.